url65a69497a4_sweider_BankSystem_JavaLab8_tgz-pJ8-banksystem_lab4_LauncherJ8/src/banksystem/lab4/core/account/AccountProxy.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * To change this license header, choose License Headers in Project Properties.
+ * To change this template file, choose Tools | Templates
+ * and open the template in the editor.
+ */
+
+package banksystem.lab4.core.account;
+
+import java.util.concurrent.Semaphore;
+
+/**
+ *
+ * @author alex
+ */
+public class AccountProxy implements IAccountProxy {
+    
+    private final Account account;
+    
+    
+    private final ITransactionAccess transactionInterface;
+    private final ICashierAccess cashierInterface;
+    private final ISummaryCheckerAccess summaryCheckerInterface;
+    
+    private final Semaphore cashierInterfaceSemaphore;
+    
+    
+    
+    public AccountProxy(Account account){
+        this.account=account;
+        this.transactionInterface = new TransactionAccess(this.account);
+        this.cashierInterface = new CashierAccess(this.account);
+        this.summaryCheckerInterface = new SummaryCheckerAccess(this.account);
+        this.cashierInterfaceSemaphore=new Semaphore(1);
+        
+    }
+    
+    @Override
+    public ITransactionAccess getTransactionInterface(){
+        return this.transactionInterface;
+    }
+    
+    @Override
+    public ICashierAccess acquireCashierInterface(){
+        try {
+            this.cashierInterfaceSemaphore.acquire();
+        } catch (InterruptedException ex) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: InterruptedException is ignored. This can delay thread shutdown and clear the thread’s interrupt status. Only code that implements a thread’s interruption policy can swallow an interruption request.

**Fix**: Rethrow the InterruptedException or reinterrupt the current thread using *Thread.currentThread().interrupt()* so that higher-level interrupt handlers can function correctly.
 If you are wrapping the InterruptedException inside a RuntimeException, call *Thread.currentThread().interrupt()* before throwing the RuntimeException.

Learn more about [interrupts](https://docs.oracle.com/javase/tutorial/essential/concurrency/interrupt.html) and [dealing with InterruptedException](https://www.ibm.com/developerworks/java/library/j-jtp05236/index.html?ca=drs-#N10187)